package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -727,3 +740,89 @@ def has_forces(self):
     @has_forces.setter
     def has_forces(self, value: bool):
         self._has['force'] = value
+
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        del state['_particle_group']
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__ = state
+        self._particle_group = self._file['particles'][
+                               list(self._file['particles'])[0]]
+        self[self.ts.frame]
+
+
+class H5PYPicklable(h5py.File):
+    """H5PY file object (read-only) that can be pickled.
+
+    This class provides a file-like object (as returned by
+    :class:`h5py.File`) that,
+    unlike standard Python file objects,
+    can be pickled. Only read mode is supported.
+
+    When the file is pickled, filename, mode, driver, and comm of
+    :class:`h5py.File` in the file are saved. On unpickling, the file
+    is opened by filename, mode, driver, and comm (driver and comm are
+    optional arguments). This means that for a successful unpickle, the
+    original file still has to be accessible with its filename.
+
+    Parameters
+    ----------
+    filename : str or file-like
+        a filename given a text or byte string.
+    driver : str (optional)
+        H5PY file driver used to open H5MD file
+    comm : :class:`MPI.Comm` (optional)
+        MPI communicator used to open H5MD file
+        Must be passed with `'mpio'` file driver
+
+    Example
+    -------
+    ::
+
+        f = H5PYPicklable('filename', 'r')
+        print(f['particles/trajectory/position/value'][0])
+        f.close()
+
+    can also be used as context manager::
+
+        with H5PYPicklable('filename', 'r'):
+            print(f['particles/trajectory/position/value'][0])
+
+    See Also
+    ---------
+    :class:`MDAnalysis.lib.picklable_file_io.FileIOPicklable`
+    :class:`MDAnalysis.lib.picklable_file_io.BufferIOPicklable`
+    :class:`MDAnalysis.lib.picklable_file_io.TextIOPicklable`
+    :class:`MDAnalysis.lib.picklable_file_io.GzipPicklable`
+    :class:`MDAnalysis.lib.picklable_file_io.BZ2Picklable`
+
+
+    .. versionadded:: 2.0.0
+    """
+    def __getstate__(self):
+        try:
+            driver = self.driver
+        except AttributeError:
+            driver = None
+
+        return {'name': self.filename,
```

Comment:
I am confused: If the `comm` is not included in `__getstate__` then how can it be used in `__setstate__`?

Reply:
I see that you had it included in b29e675765c71ee0e015b3690e6b38a902a86188 but then removed again.

Did you test that a `MPI.COMM_WORLD` could be pickled on its own, i.e.,
```python
from mpi4py import MPI
import pickle

comm = MPI.COMM_WORLD
comm_p = pickle.loads(pickle.dumps(comm))

assert comm == comm_p
```
or something similar?

Reply:
That does not seem to work
![image](https://user-images.githubusercontent.com/59893899/89835031-c50ec900-db18-11ea-9f8a-d199c2789f89.png)

Reply:
I don’t have a working mpi4py in my desktop. But it seems that it cannot be pickled natively.
This might help meanwhile https://bitbucket.org/mpi4py/mpi4py/issues/104/pickling-of-mpi-comm